source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -33,6 +33,25 @@ DnsCacheImpl::DnsCacheImpl(
       host_ttl_(PROTOBUF_GET_MS_OR_DEFAULT(config, host_ttl, 300000)),
       max_hosts_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, max_hosts, 1024)) {
   tls_slot_.set([&](Event::Dispatcher&) { return std::make_shared<ThreadLocalHostInfo>(*this); });
+
+  // Pre-Loaded hostnames are resolved without a read lock on primary hosts because it is done
+  // during object construction.
+  uint pre_loaded_hostnames{};
+  for (auto it = config.pre_load_hostnames().begin();
+       it != config.pre_load_hostnames().end() && pre_loaded_hostnames < max_hosts_;
+       ++it, ++pre_loaded_hostnames) {
+    // No need to get a resolution handle on this resolution as the only outcome needed is for the
+    // cache to load an entry. Further if this particular resolution fails all the is lost is the
+    // potential optimization of having the entry be pre-loaded the first time a true consumer of
+    // this DNS cache asks for it.
+    main_thread_dispatcher_.post([this, host = it->address(), default_port = it->port_value()]() {
+      startCacheLoad(host, default_port);
```

Comment:
wait, are we pre-loading pure addresses here?  I recall some issue about the dns cache doing dns lookups for IP addresses but can we fix this by just saying "127.0.0.3" resolves to 127.0.0.3?

Reply:
`address` can be any string here. So people could load straight up IPs or DNS names. 

The underlying dns resolvers short circuit and simply spit out the IP address if the string passed to them is a valid IP address. I fixed those issues.

Reply:
wow, I had to check API docs, but"IP (*STATIC* or *EDS* clusters) or a hostname resolved by DNS".  That's whacky.  How about switching the test to a non-address hostname since that's what it's for?